dags/solutions_team/configs/inference/inference_benchmark_config.py

```diff
@@ -0,0 +1,273 @@
+      # Install JAX and Pallas.
+      "pip install torch_xla[tpu] -f https://storage.googleapis.com/libtpu-releases/index.html",
+      "pip install torch_xla[pallas] -f https://storage.googleapis.com/jax-releases/jax_nightly_releases.html -f https://storage.googleapis.com/jax-releases/jaxlib_nightly_releases.html",
+      # Install other build dependencies.
+      "pip install setuptools-scm>=8",
+      "pip install -r requirements-tpu.txt",
+      # Build vLLM
+      'VLLM_TARGET_DEVICE="tpu" python setup.py develop',
+      # Download dataset
+      "cd .. && wget --no-verbose https://huggingface.co/datasets/anon8231489123/ShareGPT_Vicuna_unfiltered/resolve/main/ShareGPT_V3_unfiltered_cleaned_split.json",
+      # Download benchmark
+      "rm -rf ai-on-gke && git clone https://github.com/GoogleCloudPlatform/ai-on-gke",
+  )
+
+  return setup_cmds
+
+
+def get_vllm_benchmark_cmds(
+    model_id: str, num_chips: int, test_run_id: str, model_configs: Dict = {}
+):
+  base_model_id = model_id.split("/")[-1]
+  request_rates = model_configs["request_rates"].split(",")
+  instance_type = model_configs["instance_type"]
+  num_prompts = 1000
+
+  run_cmds = [
+      "export PATH=$PATH:/home/cloud-ml-auto-solutions/vllm:/home/cloud-ml-auto-solutions/.local/bin",
+      # HF_TOKEN is set in Composer environment variables
+      f"export HF_TOKEN={HF_TOKEN}",
+      # Start virtual environment
+      '[[ -f ".env/bin/activate" ]] && source .env/bin/activate',
+      # Start vllm in the background
+      f"vllm serve {model_id} --swap-space 16  --disable-log-requests --tensor_parallel_size={num_chips} --max-model-len=2048 --num-scheduler-steps=4 &",
+      # Wait for server to come up
+      "sleep 600",
+  ]
+
+  # Group metrics together using test_run_id.
+  metadata = {
+      "test_run_id": test_run_id,
+      "instance_type": instance_type,
+      "num_accelerators": num_chips,
+  }
+  for request_rate in request_rates:
+    benchmark_cmd_fmt = "python ai-on-gke/benchmarks/benchmark/tools/profile-generator/container/benchmark_serving.py --host localhost --port 8000 --num-prompts {num_prompts} --max-input-length 1024 --max-output-length 1024 --dataset ShareGPT_V3_unfiltered_cleaned_split.json --save-json-results --model '{model_id}' --tokenizer '{model_id}' --request-rate {request_rate} --additional-metadata-metrics-to-save '{additional_metadata}'"
+
+    benchmark_cmds = [
+        # Run benchmark
+        benchmark_cmd_fmt.format(
+            num_prompts=num_prompts,
+            model_id=model_id,
+            request_rate=request_rate,
+            additional_metadata=json.dumps(metadata),
+        ),
+        # Process result json files
+        f'export OUTPUT_FORMAT="*vllm*{base_model_id}*"',
+        "export BENCHMARK_OUTPUT=$(find . -name $OUTPUT_FORMAT -type f -printf \"%T@ %Tc %p\n\" | sort -n | head -1 | awk 'NF>1{print $NF}')",
+        "cat ${BENCHMARK_OUTPUT} >> metric_report.jsonl",
+        "echo '' >> metric_report.jsonl",
```

Comment:
Do the logs show up in the Airflow viewer as well? If not, it would be good to add this functionality in case it needs to be debugged

Reply:
Added one instruction to dump the file contents into task logs.